apps/archive/common.py

```diff
@@ -396,41 +399,32 @@ def remove_unwanted(doc):
+    If media files are not references by any other
+    story then delete the media files
+    :param dict doc: document for which the media are being deleted
+    :return boolean: True if files are deleted else false.
     """
-    print('Removing Media Files...')
-
-    if doc.get(ITEM_TYPE) in [CONTENT_TYPE.PICTURE, CONTENT_TYPE.VIDEO, CONTENT_TYPE.AUDIO]:
-        base_image_id = doc.get('renditions', {}).get('baseImage', {}).get('media')
-
-        if base_image_id:
-            try:
-                archive_docs = superdesk.get_resource_service('archive'). \
-                    get_from_mongo(None, {'renditions.baseImage.media': base_image_id})
-                ingest_docs = superdesk.get_resource_service('ingest'). \
-                    get_from_mongo(None, {'renditions.baseImage.media': base_image_id})
-                legal_archive_docs = superdesk.get_resource_service('legal_archive'). \
-                    get_from_mongo(None, {'renditions.baseImage.media': base_image_id})
-                archive_version_docs = superdesk.get_resource_service('archive_versions'). \
-                    get_from_mongo(None, {'renditions.baseImage.media': base_image_id})
-                legal_archive_version_docs = superdesk.get_resource_service('legal_archive_versions'). \
-                    get_from_mongo(None, {'renditions.baseImage.media': base_image_id})
-
-                if archive_docs.count() == 0 and ingest_docs.count() == 0 and legal_archive_docs.count() == 0 and \
-                        archive_version_docs.count() == 0 and legal_archive_version_docs.count() == 0:
-                    # there's no reference so do remove the file
-                    for name, rendition in doc.get('renditions').items():
-                        if 'media' in rendition:
-                            print('Deleting media:{}'.format(rendition.get('media')))
-                            app.media.delete(rendition.get('media'))
-                    # files are removed
-                    return True
-            except Exception as ex:
-                logger.exception(ex)
-                print('Removing Media Exception:{}'.format(ex))
-                return False
-    return False
+    logger.info('Removing Media Files...')
+
+    renditions = doc.get('renditions', {})
+    if not renditions:
+        renditions = (doc.get(ASSOCIATIONS) or {}).get('featuremedia', {}).get('renditions', {})
+
+    if not renditions:
+        return
+
+    for rendition in renditions.values():
+        media = rendition.get('media') if isinstance(rendition.get('media'), str) else str(rendition.get('media'))
+        try:
+            references = get_resource_service('media_references').get(req=None, lookup={
+                'media_id': media, 'published': True
+            })
+
+            if references.count() == 0:
+                logger.info('Deleting media:{}'.format(rendition.get('media')))
+                app.media.delete(media)
```

Comment:
If media is not published via any of its item it will remove the delete the media file.